src/http/index.ts

```diff
@@ -87,16 +90,91 @@ app.get("/data/:chainId/rounds/:roundId/applications.csv", async (req, res) => {
       application.metadata.application.project.projectTwitter,
       application.metadata.application.project.projectGithub,
       application.metadata.application.project.userGithub,
-      ...answers,
+      ...answers
     ]);
   }
 
   res.setHeader("content-type", "text/csv");
   res.send(csv.getHeaderString() + csv.stringifyRecords(records));
 });
 
+app.get("/data/:chainId/rounds/:roundId/vote_coefficients.csv", async (req, res) => {
+  const db = loadDatabase(req.params.chainId);
+
+  const votes = await db
+    .collection(`rounds/${req.params.roundId}/votes`)
+    .all();
+
+  const data = fs.readFileSync("./data/passport_scores.json", {
+    encoding: "utf8",
+    flag: "r"
+  });
+
+  /*TODO: type this once we have unified types*/
+  const passportScores: any[] = JSON.parse(data);
+
+  const processPassportScores = (scores: any[]): Record<string, any> => {
+    return scores.reduce((map: Record<string, any>, score: any) => {
+      const address = score.address.toLowerCase();
+      const { evidence, error, ...remainingScore } = score;
+      map[address] = { ...remainingScore, ...evidence };
+      return map;
+    }, {});
+  };
+
+  const passportScoresMap = processPassportScores(passportScores);
+
+  const voteCoefficients: any[] = votes.map((vote: any) => {
+    const voter = vote.voter.toLowerCase();
+    return { ...vote, ...passportScoresMap[voter] ?? {} };
+  });
+
+  const records = voteCoefficients.map((voteCoefficient) => [
```

Comment:
We're still unnecessarily looping through the array twice, once to join with passport and the second time to convert it to a CSV row, we should do this in one pass

Reply:
@boudra @hmrtn @vacekj 

Maybe we can improve it even more if instead of loading `passport_scores.json`, we load `passport_valid_addresses.json` directly, so we don't need to loop and check the coefficients, they are already the smaller subset of valid addresses.

We can modify `passport/index.ts` to lowercase the addresses, so we don't need to do it again here. 

We can modify `passport/index.ts` to store an object instead of an array, so we load it and it's already in the final shape we want.

We can try with different data structures to see which one has the fastest lookup among Object, Set, and Map.
I benchmarked with a set of 9 millions strings of 42 characters (like eth addresses), and the average lookups with those data structures are:

```
avgSet 0.0011645 ms
avgMap 0.00043525 ms
avgObj 0.000993375 ms
```

wdyt?

Reply:
@gravityblast The `passport_valid_addresses.json` approach is a good idea, and from the quick test it appears Map is the fastest. Alternatively, we could cache the passportScoresMap to avoid recalculating it on every request - if we expect it to not be updated too often. 

@boudra It will still be linear but this is true - we can further optimize this to a single map!

Reply:
I like the idea of writing an object directly! Maybe we could even just store the threshold as the value instead of all that extra data now that we know it's what we need.

But for me this is now good enough to merge if we want to do this later, together with the calculator code that also accesses this data. 

My intention here was only to make sure we're not pushing very resource intensive code unnecessarily, not really optimizing.

Reply:
yes let's try my optimization in another PR!

but we need to fix the loop in this PR to avoid looping more than once.

Reply:
this was resolved, as far as I could tell, in the latest commit!

Reply:
yeah it looks good! @hmrtn